respokeSDK/src/main/java/com/digium/respokesdk/RespokeClient.java

```diff
@@ -513,6 +533,187 @@ public RespokeGroup getGroup(String groupIDToFind) {
 
 
     /**
+     * Retrieve the history of messages that have been persisted for 1 or more groups. Only those
+     * messages that have been marked to be persisted when sent will show up in the history. Only
+     * the most recent message in each group will be retrieved - to pull more messages, use the
+     * other method signature that allows `maxMessages` to be specified.
+     *
+     * @param groupIds The groups to pull history for
+     * @param completionListener The callback called when this async operation has completed
+     */
+    public void getGroupHistories(ArrayList<String> groupIds,
+                                  final GroupHistoriesCompletionListener completionListener) {
+        getGroupHistories(groupIds, 1, completionListener);
+    }
+
+    /**
+     * Retrieve the history of messages that have been persisted for 1 or more groups. Only those
+     * messages that have been marked to be persisted when sent will show up in the history.
+     *
+     * @param groupIds The groups to pull history for
+     * @param maxMessages The maximum number of messages per group to pull. Must be >= 1
+     * @param completionListener The callback called when this async operation has completed
+     */
+    public void getGroupHistories(final ArrayList<String> groupIds, final Integer maxMessages,
```

Comment:
I would use a List here. Users of the client may build lists using different implementations (ArrayList, LinkedList...).